src/Orleans.Core/Providers/LegacyProviderConfigurator.cs

```diff
@@ -0,0 +1,168 @@
+            private readonly int defaultInitStage;
+            private int initStage;
+            private readonly int defaultStartStage;
+            private int startStage;
+
+            public ProviderLifecycleParticipant(IProviderConfiguration config, IServiceProvider services, ILoggerFactory loggerFactory, int defaultInitStage, int defaultStartStage)
+            {
+                this.logger = loggerFactory.CreateLogger($"{config.Type}-{config.Name}");
+                this.services = services;
+                this.config = config;
+                this.defaultInitStage = defaultInitStage;
+                this.defaultStartStage = defaultStartStage;
+                this.schedualer = services.GetService<LegacyProviderConfigurator.QueueTask>();
+            }
+
+            public virtual void Participate(TLifeCycle lifecycle)
+            {
+                this.initStage = this.config.GetIntProperty(LegacyProviderConfigurator.InitStageName, this.defaultInitStage);
+                lifecycle.Subscribe(this.initStage, Init, Close);
+                this.startStage = this.config.GetIntProperty(LegacyProviderConfigurator.StartStageName, this.defaultStartStage);
+                lifecycle.Subscribe(this.startStage, Start);
+
+            }
+
+            private async Task Init(CancellationToken ct)
+            {
+                var stopWatch = Stopwatch.StartNew();
+                IProvider provider = this.services.GetServiceByName<TService>(this.config.Name) as IProvider;
+                IProviderRuntime runtime = this.services.GetRequiredService<IProviderRuntime>();
+                await schedual(() => provider.Init(this.config.Name, runtime, this.config));
+                stopWatch.Stop();
+                this.logger.Info(ErrorCode.SiloStartPerfMeasure, $"Initializing in stage {this.initStage} took {stopWatch.ElapsedMilliseconds} Milliseconds");
+            }
+
+            private async Task Close(CancellationToken ct)
+            {
+                var stopWatch = Stopwatch.StartNew();
+                IProvider provider = this.services.GetServiceByName<TService>(this.config.Name) as IProvider;
+                await schedual(() => provider.Close());
+                stopWatch.Stop();
+                this.logger.Info(ErrorCode.SiloStartPerfMeasure, $"Closing in stage {this.initStage} took {stopWatch.ElapsedMilliseconds} Milliseconds");
+            }
+
+            private async Task Start(CancellationToken ct)
+            {
+                var stopWatch = Stopwatch.StartNew();
+                IStreamProviderImpl provider = this.services.GetServiceByName<TService>(this.config.Name) as IStreamProviderImpl;
+                if (provider == null) return;
+                await schedual(() => provider.Start());
+                stopWatch.Stop();
+                this.logger.Info(ErrorCode.SiloStartPerfMeasure, $"Starting in stage {this.startStage} took {stopWatch.ElapsedMilliseconds} Milliseconds");
```

Comment:
Provider name is already part of the logger name.

Reply:
sorry I wasn't clear, I referred logger name as category name. Since not all ILogger implementation prints out category/logger name, so it maybe a good idea to print the provider name since it is useful in debugging.